layer.go

```diff
@@ -67,46 +69,49 @@ type Layer struct {
 	// finalHandler stores the final middleware chain handler.
 	finalHandler http.Handler
 
-	// memo stores the memoized middleware call chain by specific phase.
-	memo map[string]http.Handler
```

Comment:
I did some benchmarks and the performance benefit of memoizing at layer level has no significant benefit. Also, it adds some code in terms of flushing memoized value, for instance, when a new parent layer is attached. We can support memoization in the future, when it makes more sense, but I can ensure that the layer won't be a limiter in terms of performance.